eventlet/hubs/__init__.py

```diff
@@ -71,36 +64,31 @@ def use_hub(mod=None):
         mod = get_default_hub()
     if hasattr(_threadlocal, 'hub'):
         del _threadlocal.hub
+
+    classname = ''
     if isinstance(mod, six.string_types):
         assert mod.strip(), "Need to specify a hub"
         if '.' in mod or ':' in mod:
             modulename, _, classname = mod.strip().partition(':')
-            mod = __import__(modulename, globals(), locals(), [classname])
-            if classname:
-                mod = getattr(mod, classname)
         else:
-            found = False
-
-            # setuptools 5.4.1 test_import_patched_defaults fail
-            # https://github.com/eventlet/eventlet/issues/177
-            try:
-                # try and import pkg_resources ...
-                import pkg_resources
-            except ImportError:
-                # ... but do not depend on it
-                pkg_resources = None
-            if pkg_resources is not None:
-                for entry in pkg_resources.iter_entry_points(
-                        group='eventlet.hubs', name=mod):
-                    mod, found = entry.load(), True
-                    break
-            if not found:
-                mod = __import__(
-                    'eventlet.hubs.' + mod, globals(), locals(), ['Hub'])
-    if hasattr(mod, 'Hub'):
-        _threadlocal.Hub = mod.Hub
+            modulename = 'eventlet.hubs.' + mod
+        mod = importlib.import_module(modulename)
+
+    if hasattr(mod, 'is_available'):
```

Comment:
No, it catches `AttributeError` inside `is_available()` call. I try to avoid hiding/changing errors when possible. Same code would look like this:

```
check = lambda: True
try:
  check = mod.is_available
except AttributeError:
  ...
if not check():
  raise HubError(...)
```

Please, take no offense, I deeply regret that in most Python code `exception` lost the meaning of the word.